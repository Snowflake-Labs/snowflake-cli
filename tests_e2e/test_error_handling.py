import subprocess

import pytest


@pytest.mark.e2e
def test_error_traceback_disabled_without_debug(snowcli, test_root_path):
    traceback_msg = "Traceback (most recent call last)"
    import os  # TODO remove thise line and three below
    import stat
    READABLE_BY_OTHERS = stat.S_IRGRP | stat.S_IROTH
    print(f"*******FILE CHECK: {os.stat(test_root_path / 'config' / 'malformatted_config.toml') & READABLE_BY_OTHERS}")
    result = subprocess.run(
        [
            snowcli,
            "--config-file",
            test_root_path / "config" / "malformatted_config.toml",
            "sql",
            "-q",
            "select 'Hello there'",
        ],
        capture_output=True,
        text=True,
    )
    assert result.stderr == "" and not traceback_msg in result.stdout

    result_debug = subprocess.run(
        [
            snowcli,
            "--config-file",
            test_root_path / "config" / "malformatted_config.toml",
            "sql",
            "-q",
            "select 'Hello there'",
            "--debug",
        ],
        capture_output=True,
        text=True,
    )
    assert traceback_msg in result_debug.stderr
