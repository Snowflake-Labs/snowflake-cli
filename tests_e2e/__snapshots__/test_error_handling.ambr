# serializer version: 1
# name: test_corrupted_config_in_default_location
  '''
  ╭─ Error ──────────────────────────────────────────────────────────────────────╮
  │ Configuration file seems to be corrupted. Key "demo" already exists. at line │
  │ 2 col 18                                                                     │
  ╰──────────────────────────────────────────────────────────────────────────────╯
  
  '''
# ---
# name: test_corrupted_config_in_default_location.1
  '''
  +-----------------------------------------------------+
  | connection_name | parameters           | is_default |
  |-----------------+----------------------+------------|
  | dev             | {}                   | False      |
  | integration     | {'schema': 'public'} | False      |
  +-----------------------------------------------------+
  
  '''
# ---
